### ISSUE TITLE
2.x does not compile on solaris

### ISSUE THREAD

**Issue by edgargabriel** [open-mpi/ompi#1328]
As reported by Siegmar Gross on the mailing list, the 2.x branch does not compile on Solaris because of non-portable usage of statfs

```
log.make.SunOS.sparc.64_cc
   CCLD     mca_fs_ufs.la
make[2]: Leaving directory 
`/export2/src/openmpi-2.0.0/openmpi-v2.x-dev-1020-ge2a53b3-SunOS.sparc.64_cc/ompi/mca/fs/ufs'
Making all in mca/io/ompio
make[2]: Entering directory 
`/export2/src/openmpi-2.0.0/openmpi-v2.x-dev-1020-ge2a53b3-SunOS.sparc.64_cc/ompi/mca/io/ompio'
   CC       io_ompio.lo
   CC       io_ompio_component.lo
"../../../../../openmpi-v2.x-dev-1020-ge2a53b3/ompi/mca/io/ompio/io_ompio_component.c", 
line 285: prototype mismatch: 2 args passed, 4 expected
"../../../../../openmpi-v2.x-dev-1020-ge2a53b3/ompi/mca/io/ompio/io_ompio_component.c", 
line 290: prototype mismatch: 2 args passed, 4 expected
"../../../../../openmpi-v2.x-dev-1020-ge2a53b3/ompi/mca/io/ompio/io_ompio_component.c", 
line 296: undefined struct/union member: f_type
cc: acomp failed for 
../../../../../openmpi-v2.x-dev-1020-ge2a53b3/ompi/mca/io/ompio/io_ompio_component.c
make[2]: *** [io_ompio_component.lo] Error 1
make[2]: Leaving directory 
`/export2/src/openmpi-2.0.0/openmpi-v2.x-dev-1020-ge2a53b3-SunOS.sparc.64_cc/ompi/mca/io/ompio'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory 
`/export2/src/openmpi-2.0.0/openmpi-v2.x-dev-1020-ge2a53b3-SunOS.sparc.64_cc/ompi'
make: *** [all-recursive] Error 1
tyr openmpi-v2.x-dev-1020-ge2a53b3-SunOS.sparc.64_cc 34
```

***EDIT:*** Adding triple-tick quoting for Edgar's paste of Siegmar's compile to make it a bit easier to read.

**Comment by jsquyres** [comment]
Adding @siegmargross.

**Comment by edgargabriel** [comment]
fixed with pr #1331 and pr#935 on ompi-release
